api/models/Group.js

```diff
@@ -880,29 +881,58 @@ module.exports = bookshelf.Model.extend(merge({
     }).query()
   },
 
-  async selectIdsForModeratedGroups (userId) { // TODO RESP: .... this was the last thing I was working on before getting sick and then going to kiwiburn... so I don't recal whether its worth keeping around or not...
-    // get all responsiblities for a user
-    const responsibilities = await Responsibility.fetchSystemResponsiblititesForUser(userId)
-    const byGroupId = {}
-    const result = []
-    // count how many responsibilities a user has for each group
-    responsibilities.forEach(r => {
-      if (byGroupId[r.group_id]) {
-        byGroupId[r.group_id] += 1
-      } else {
-        byGroupId[r.group_id] = 1
-      }
+  selectIdsByResponsibilities (userOrId, responsibilities) {
+    const throughCommonRole = MemberCommonRole.query(q => {
+      q.select('group_id')
+      whereId(q, userOrId, 'group_memberships_common_roles.user_id')
+      q.join('common_roles_responsibilities', 'common_roles_responsibilities.common_role_id', 'group_memberships_common_roles.common_role_id')
+      q.where('common_roles_responsibilities.responsibility_id', 'in', responsibilities)
     })
 
-    // return an array of group ids that the user has 4 responsibilities for. This defacto means they have full power of a group, whether its from a manager role or not
-    for (const key in byGroupId) {
-      if (byGroupId[key] === 4) {
-        result.push(key)
-      }
-    }
-    return result
+    const throughGroupRole = MemberGroupRole.collection().query(q => {
+      q.select('group_id')
+      whereId(q, userOrId, 'group_memberships_group_roles.user_id')
+      q.join('group_roles_responsibilities', 'group_roles_responsibilities.group_role_id', 'group_memberships_group_roles.group_role_id')
+      q.where('group_roles_responsibilities.responsibility_id', 'in', responsibilities)
+    })
+
+    return GroupMembership.forIds(userOrId, null, {
+      query: q => {
+        q.select('groups.id')
+        q.join('groups', 'groups.id', 'group_memberships.group_id')
+        q.where('groups.active', true)
+        q.where((q2) => {
+          q2.whereIn('groups.id', throughCommonRole.query())
+          q2.orWhereIn('groups.id', throughGroupRole.query())
+        })
+      },
+      multiple: true
+    }).query()
   },
 
+  // async selectIdsForModeratedGroups (userId) { // TODO RESP: .... this was the last thing I was working on before getting sick and then going to kiwiburn... so I don't recal whether its worth keeping around or not...
```

Comment:
you can just nuke this if it wasn't apparent where it was needed

Reply:
yeah I found an alternative way